### ISSUE TITLE
Feature Request: :eq(index) selector

### ISSUE THREAD

**Issue by arcanis** [ericchiang/pup#39]
Hi,

Could it be possible to implement the [jQuery :eq() selector](https://api.jquery.com/eq-selector/)? Sometimes :nth-child is not enough to select the right element :)

**Comment by ericchiang** [comment]
Seems reasonable. I'll add it when I get a sec.

**Comment by danger89** [comment]
I really like to have this! Any progress so far??

For example I'm parsing a table. When I do ```pup 'tr td'``` I see a list of table cells, which is expected. But when I try to receive the number of a td in this case, :eq(n) or :index(n) can be very useful! :nth-child.. doesn't always work as expected.

**Comment by jonhoo** [comment]
Is `:nth-of-type` not sufficient?

**Comment by Ghoughpteighbteau** [comment]
I've not his a point where nth child doesn't work. eq has some pretty strange behavior, in that it returns the first result only, rather than what matches. Theoretically I think it's kinda useless, but maybe in practice it's got some application?

**Comment by danger89** [comment]
Yeah you are right. I did it wrong. When I did a nth-of-type on a 'td'. I got 2 results, I expected one. But the issue was that I also needed a nth-of-type on a 'tr' as well to make the result more explicit. Like so:
```table tr:nth-of-type(2) td:nth-of-type(1)```

**Comment by jonhoo** [comment]
I'm pretty sure that problem should go away if you use child selectors (`>`).
